File: 13 лаба/13A.py

fin = open("search1.in",  "r")
fout = open("search1.out", "w")


pattern = fin.readline()
text = fin.readline()

M = len(pattern)
N = len(text)

places = []
count = 0

for i in range(N-M+2):
	if text[i:(i+M-1)] == pattern[0:-1]:
		places.append(i+1)
		count+=1


fout.write(str(count))
fout.write("\n")
for i in range(len(places)): 
	fout.write(str(places[i]))
	fout.write(" ")


fin.close()
fout.close()

File: 13 lab - Substring search/13A.cpp
#include <fstream>
#include <vector>

using namespace std;

int main() {

    ifstream fin("search1.in");
    ofstream fout("search1.out");

    string text, pattern;
    int start;
    int count = 0;
    bool check;


    fin >> pattern >> text;

    vector<int> places;

    for(int i=0; i < (text.length() - pattern.length()+1);i++){

        if (text[i]==pattern[0]){
            start = i;
            check = true;

            for (int j=1; j < pattern.length(); j++){

                i+=1;

                if (text[i]!=pattern[j]){

                    check = false;
                    break;
                }

            }
            if (check==true){

                places.push_back(start+1);
                count+=1;
            }
            i = start;
        }
    }
    fout << count << endl;
    for (int i=0; i< places.size(); i++){
        fout << places[i] << " ";
    }

    return 0;
}
